File: 2020-09-12 - Complejidad algorítmica/live/s1.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
  int n;
  cin >> n;
  vector<int> a(n);
  for (int i = 0; i < n; i++) {
    cin >> a[i];
  }

  // O(n^2)
  for (int p = 0; p < n; p++) { // O(n)
    int left = 0, right = 0;
    // left = a[0] + a[1] + ... + a[p-1]
    for (int i = 0; i <= p - 1; i++) { // O(n)
      left += a[i]; // O(1)
    }
    // right = a[p] + a[p+1] + a[p+2] + ... + a[n-1]
    for (int i = p; i <= n - 1; i++) { // O(n)
      right += a[i]; // O(1)
    }

    if (left > 0 and right < 0) {
      cout << p << endl;
      return 0;
    }
  }
  cout << "Impossible" << endl;
  return 0;
}

File: 2020-09-12 - Complejidad algorítmica/live/s1.py
def solve():
  n = int(input())
  a = [int(x) for x in input().split()]

  # O(n^2)
  for p in range(n): # O(n)
    left = sum(a[0:p]) # sumar a[0] a a[p-1]  # O(n)
    right = sum(a[p:n]) # sumar a[p] a a[n-1] # O(n)
    if left > 0 and right < 0: # O(1)
      print(p) # O(1)
      return   # O(1)
  print("Impossible")

if __name__ == "__main__":
  solve()